packages/pyright-internal/src/analyzer/parseTreeUtils.ts

```diff
@@ -96,25 +96,46 @@ export function findNodeByOffset(node: ParseNode, offset: number): ParseNode | u
 
     // The range is found within this node. See if we can localize it
     // further by checking its children.
-    const children = getChildNodes(node);
+    let children = getChildNodes(node);
+    if (isCompliantWithNodeRangeRules(node) && children.length > 20) {
+        // use Binary search to find child to visit. This should help
```

Comment:
Nit: "Use" and "Otherwise" should be capitalized because they start a sentence. "Binary" should not. "sibling" should be singular.